framework/src/base/NonlinearSystem.C

```diff
@@ -1253,6 +1254,12 @@ NonlinearSystem::computeResidualInternal(Moose::KernelType type)
       const std::vector<MooseSharedPointer<ScalarKernel> > & scalars = _scalar_kernels.getActiveObjects();
       for (const auto & scalar_kernel : scalars)
       {
+        // Don't compute the residual if this is an ODETimeKernel and
+        // we've been asked for KT_NONTIME residuals.
+        ODETimeKernel * ode_time = dynamic_cast<ODETimeKernel *>(scalar_kernel.get());
+        if (type == Moose::KT_NONTIME && ode_time)
+          continue;
+
```

Comment:
Maybe we should consider doing the same deal with 3 warehouses i.e. one warehouse for each `ALL`, `TIME` and `NONTIME` scalar kernels like we do for normal kernels.

This implementation is not a big when people have just a few scalar kernels, but I am not sure what that will do in RELAP-7 where there is going to be possibly hundreds of them.